app/src/main/java/com/example/geomate/ui/screens/groups/GroupsViewModel.kt

```diff
@@ -19,20 +21,52 @@ class GroupsViewModel(
     private val _uiState = MutableStateFlow(GroupsUiState())
     val uiState: StateFlow<GroupsUiState> = _uiState.asStateFlow()
 
+    private val _matchGroups: MutableStateFlow<MutableMap<Group, List<Uri>>> =
+        MutableStateFlow(mutableMapOf())
+    val matchGroups: StateFlow<MutableMap<Group, List<Uri>>> = _matchGroups.asStateFlow()
     fun fetchGroups(userId: String) = viewModelScope.launch {
         groupsRepository.getAll(userId).collect { groups ->
             _uiState.update {
                 it.copy(
                     groups = groups.associateWith { group ->
-                        List(group.users.size) { Uri.EMPTY }
+                        List(group.users.size) {
+                            Uri.EMPTY
+                        }
                     }.toMutableMap()
                 )
             }
+            updateMatchGroups(_uiState.value.groups)
```

Comment:
Use `groups` as a parameter for this method. `_uiState.value.groups` might not be updated yet.

Reply:
During review idea was found, that there was no need in mehtod call updateMatchGroups(...) in fetchGroups(...)